trillian/migrillian/election/election.go

```diff
@@ -22,36 +22,43 @@ import "context"
 // Election controls an instance's participation in master election process.
 // Note: Implementations are not intended to be thread-safe.
 type Election interface {
-	// Await blocks until the instance captures mastership. Returns a "mastership
-	// context" which remains active until the instance stops being the master,
-	// or the passed in context is canceled. Returns an error if capturing fails,
-	// or the passed in context is canceled before mastership is captured.
-	//
-	// Await is safe to be called again when the current mastership context is
-	// canceled, which might happen implicitly when mastership is overtaken, or
-	// explicitly when Resign is called.
+	// Await blocks until the instance captures mastership. Returns immediately
+	// if it is already the master. Returns an error if capturing fails, or the
+	// passed in context is canceled before mastership is captured. If an error
+	// is returned, the instance might still have become the master. Idempotent,
+	// might be useful to retry in case of an error.
+	Await(ctx context.Context) error
+
+	// Observe returns a "mastership context" which remains active until the
```

Comment:
What happens if `Observe()` is called without a preceding `Await()` ?  

Or is that not allowed?  (In which case, maybe old-`Await` shouldn't be split up -- a user that doesn't want to observe mastership changes can just drop the returned mastership context).

Reply:
Calling `Observe` without a preceding `Await` (or after `Await+Resign`) is okay, it will simply result in an "outright canceled context". Extended the comment to make this clearer.

Reply:
I still wonder if it's worth exposing the separate `Observe` entrypoint, if most users are always going to do `Await();Observe()` (and those that just want a one-shot "am I master right now?" can drop the mastership context).  Are there situations when `Observe` is likely to happen away from `Await` ?